### PR TITLE
Fix Xcode 12.5 support

### PR DESCRIPTION
**Pull request by dcalhoun**
## Description
There have been a few issue building React Native in Xcode 12.5. This applies a temporary fix via `patch-package` until we are able upgrade to `react-native` >=0.64.1.

- [Example Issue](https://git.io/J35zb)
- [0.64.1 Release Notes](https://git.io/J35zN)
- [Xcode 12.5 Troubleshooting Guide](https://git.io/J35zp)
- [Temporary Fix Applied](https://git.io/J35zj)

## How has this been tested?
Built, with both Xcode 12.5 and the RN CLI, and ran the mobile Demo app to verify the previous crashes (below) did not occur and no new crashes occurred. 

<details>
<summary>Xcode 12.5 build error</summary>

xcode-12 5-error

</details>

<details>
<summary>RN CLI build error</summary>

```
/Users/davidcalhoun/Sites/a8c/gutenberg-mobile/gutenberg/packages/react-native-editor/ios/Pods/Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.5.99. (in target 'boost-for-react-native' from project 'Pods')
/Users/davidcalhoun/Sites/a8c/gutenberg-mobile/gutenberg/packages/react-native-editor/ios/Pods/Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.5.99. (in target 'react-native-video' from project 'Pods')
/Users/davidcalhoun/Sites/a8c/gutenberg-mobile/gutenberg/packages/react-native-editor/ios/Pods/Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.5.99. (in target 'react-native-safe-area' from project 'Pods')
/Users/davidcalhoun/Sites/a8c/gutenberg-mobile/gutenberg/packages/react-native-editor/ios/Pods/Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.5.99. (in target 'react-native-keyboard-aware-scroll-view' from project 'Pods')
/Users/davidcalhoun/Sites/a8c/gutenberg-mobile/gutenberg/packages/react-native-editor/ios/Pods/Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.5.99. (in target 'react-native-blur' from project 'Pods')
** BUILD FAILED **
The following build commands failed:
        CompileC /Users/davidcalhoun/Sites/a8c/gutenberg-mobile/gutenberg/packages/react-native-editor/ios/build/GutenbergDemo/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/React-Core.build/Objects-normal/x86_64/RCTCxxBridge.o /Users/davidcalhoun/Sites/a8c/gutenberg-mobile/gutenberg/node_modules/react-native/React/CxxBridge/RCTCxxBridge.mm normal x86_64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler
(1 failure)
```

</details>

## Screenshots
n/a

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Review by antonis — APPROVED** [review]
Great work @dcalhoun 👍 
I confirm that I was able to run the `GutenbergDemo` app with [Xcode 12.5](https://user-images.githubusercontent.com/304044/118020933-5ee2fb00-b363-11eb-84c5-8dec83d5e2f3.png) as described 🎉 

I also tried to build the `WordPress-iOS` app locally with `LOCAL_GUTENBERG` but run into the [same issue](https://user-images.githubusercontent.com/304044/118020169-75d51d80-b362-11eb-8d71-140a1773d72a.png). Probably I need some extra configuration for this. In any case this PR is an improvement to our workflow. Thank you 🙇

**Comment by dcalhoun** [comment]
> I also tried to build the `WordPress-iOS` app locally with `LOCAL_GUTENBERG` but run into the [same issue](https://user-images.githubusercontent.com/304044/118020169-75d51d80-b362-11eb-8d71-140a1773d72a.png). Probably I need some extra configuration for this. 

Ah, interesting. Thank you for calling out this issue. I will explore what might be happening.

**Comment by dcalhoun** [comment]
> I also tried to build the `WordPress-iOS` app locally with `LOCAL_GUTENBERG` but run into the [same issue](https://user-images.githubusercontent.com/304044/118020169-75d51d80-b362-11eb-8d71-140a1773d72a.png). Probably I need some extra configuration for this. In any case this PR is an improvement to our workflow. Thank you 🙇

@antonis I tested building WPiOS with this branch and it succeeded. I'm not sure what issue you may have run into. Maybe `npm install` was not run after switching `gutenberg-mobile` branches? Running `npm install` is required for the patches in this PR to be applied to React Native in the project's `node_modules` directory, otherwise the build errors would continue to occur. 

I am going to merge this work, but let me know if you have any specifics on the error or continue to see issues with Xcode 12.5. Thanks!

**Comment by ceyhun** [comment]
Hey there @dcalhoun 👋 I'm trying to apply the patch to RN 0.64.0 upgrade [here](https://github.com/WordPress/gutenberg/pull/29118), after getting a merge from `trunk`. But I think these changes are already fixed on 0.64.0:

- https://github.com/facebook/react-native/blob/v0.64.0/React/CxxBridge/RCTCxxBridge.mm

- https://github.com/facebook/react-native/blob/v0.64.0/ReactCommon/react/nativemodule/core/platform/ios/RCTTurboModuleManager.mm

In the 0.64.1 changelog I saw fixes related to Flipper (which we won't be including in the 0.64.0 upgrade) but not these 2 files. I still haven't tried compiling 0.64 branch with Xcode 12.5, but if it compiles fine, do you think it would be safe to remove this patch from 0.64.0 upgrade?

**Comment by dcalhoun** [comment]
> I still haven't tried compiling 0.64 branch with Xcode 12.5, but if it compiles fine, do you think it would be safe to remove this patch from 0.64.0 upgrade?

@ceyhun yes, if the project will successfully build in Xcode 12.5, then I believe we can safely remove these patch files. Thank you!

**Pull request merged**
